Applications/Spire/Include/Spire/Ui/KeySequenceInputField.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef SPIRE_KEY_SEQUENCE_INPUT_FIELD_HPP
+#define SPIRE_KEY_SEQUENCE_INPUT_FIELD_HPP
+#include <QKeySequence>
+#include <QLineEdit>
+#include <QPainter>
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Reprents an input for key sequences.
+  class KeySequenceInputField : public QLineEdit {
+    public:
+
+      //! Constructs a KeySequenceInputField.
+      /*!
+        \param model The associated validation model.
+        \param parent The parent widget.
+      */
+      KeySequenceInputField(std::shared_ptr<KeySequenceValidationModel> model,
+        QWidget* parent = nullptr);
+
+      //! Adds a key to the list of unvalidated input keys.
+      /*
+        \param key The key to add.
+      */
+      void add_key(Qt::Key key);
```

Comment:
What is this?

Reply:
Because the escape key would close the editor when used in a table, I needed a way to filter the escape key in the delegate and forward it to the key sequence editor. This was my initial solution when I implemented the Key Bindings table, but I may be able to use either postEvent or sendEvent instead.

Reply:
Prefer one of post/send event.